cpp/tests/context.cpp

```diff
@@ -19,23 +27,29 @@ TEST(ContextTest, DefaultConfigsAndFlags)
 {
   static constexpr auto featureFlags = ucxx::Context::defaultFeatureFlags;
   auto context                       = ucxx::createContext({}, featureFlags);
-
-  auto configMapOut = context->getConfig();
+  auto configMapOut                  = context->getConfig();
   ASSERT_GT(configMapOut.size(), 1u);
   ASSERT_NE(configMapOut.find("TLS"), configMapOut.end());
-  ASSERT_EQ(configMapOut["TLS"], "all");
+  if (const char* envTls = std::getenv("UCX_TLS"))
+    ASSERT_EQ(configMapOut["TLS"], envTls);
+  else
+    ASSERT_EQ(configMapOut["TLS"], "all");
 
   ASSERT_EQ(context->getFeatureFlags(), featureFlags);
 }
 
```

Comment:
Does it make sense to test the corner cases for the parsing properly? I don't know this test system well, but a naive thing would be to write:

```
  std::vector<std::pair<std::string_view, bool>> cases{{"all", true},
                                                       {"cuda", true},
                                                       {"cuda_copy", true},
                                                       {"rc,cuda", true},
                                                       {"^rc,cuda_ipc", true},
                                                       {"tcp,rc", false},
                                                       {"^rc,tcp,cuda", false},
                                                       {"^cuda_copy", false},
                                                       {"^rc,cuda_copy", false}};
  for (auto&& [v, expect]: cases) {
    ASSERT_EQ(has_support(v), expect);
  }
```
if the has_support call were exposed. I guess otherwise one creates the context and then checks for `_cudaSupport`?

Reply:
Ah, OK, makes sense.